pandas/tests/io/conftest.py

```diff
@@ -50,93 +41,71 @@ def xml_file(datapath):
     return datapath("io", "data", "xml", "books.xml")
 
 
-@pytest.fixture
-def s3_base(worker_id, monkeypatch):
-    """
-    Fixture for mocking S3 interaction.
+@pytest.fixture(scope="session")
+def aws_credentials(monkeysession):
+    """Mocked AWS Credentials for moto."""
+    monkeysession.setenv("AWS_ACCESS_KEY_ID", "testing")
+    monkeysession.setenv("AWS_SECRET_ACCESS_KEY", "testing")
+    monkeysession.setenv("AWS_SECURITY_TOKEN", "testing")
+    monkeysession.setenv("AWS_SESSION_AWS_SESSION_TOKEN", "testing")
+    monkeysession.setenv("AWS_DEFAULT_REGION", "us-east-1")
 
-    Sets up moto server in separate process locally
-    Return url for motoserver/moto CI service
-    """
-    pytest.importorskip("s3fs")
-    pytest.importorskip("boto3")
-
-    # temporary workaround as moto fails for botocore >= 1.11 otherwise,
-    # see https://github.com/spulec/moto/issues/1924 & 1952
-    monkeypatch.setenv("AWS_ACCESS_KEY_ID", "foobar_key")
-    monkeypatch.setenv("AWS_SECRET_ACCESS_KEY", "foobar_secret")
-    if is_ci_environment():
-        if is_platform_arm() or is_platform_mac() or is_platform_windows():
-            # NOT RUN on Windows/macOS, only Ubuntu
-            # - subprocess in CI can cause timeouts
-            # - GitHub Actions do not support
-            #   container services for the above OSs
-            pytest.skip(
-                "S3 tests do not have a corresponding service on "
-                "Windows or macOS platforms"
-            )
-        else:
-            # set in .github/workflows/unit-tests.yml
-            yield "http://localhost:5000"
-    else:
-        requests = pytest.importorskip("requests")
-        pytest.importorskip("moto")
-        pytest.importorskip("flask")  # server mode needs flask too
-
-        # Launching moto in server mode, i.e., as a separate process
-        # with an S3 endpoint on localhost
-
-        worker_id = "5" if worker_id == "master" else worker_id.lstrip("gw")
-        endpoint_port = f"555{worker_id}"
-        endpoint_uri = f"http://127.0.0.1:{endpoint_port}/"
-
-        # pipe to null to avoid logging in terminal
-        with subprocess.Popen(
-            shlex.split(f"moto_server s3 -p {endpoint_port}"),
-            stdout=subprocess.DEVNULL,
-            stderr=subprocess.DEVNULL,
-        ) as proc:
-            timeout = 5
-            while timeout > 0:
-                try:
-                    # OK to go once server is accepting connections
-                    r = requests.get(endpoint_uri)
-                    if r.ok:
-                        break
-                except Exception:
-                    pass
-                timeout -= 0.1
-                time.sleep(0.1)
-            yield endpoint_uri
-
-            proc.terminate()
+
+@pytest.fixture(scope="session")
+def moto_server(aws_credentials):
+    moto_server = pytest.importorskip("moto.server")
+    server = moto_server.ThreadedMotoServer(port=0)
```

Comment:
I would prefer to continue to use the moto container in the CI since it's kept these S3 tests stable.

I would be OK using this for "local" runs.